### PR TITLE
Add size-based condition to the index rollover API

### LINKED ISSUE #27004: Add size-based check to rollover API
Initially, we added just a doc-count and date-based check to the rollover API.  We avoided a size-based check because it could be wildly inaccurate if multiple merges are in progress on different shards of the index.

We could make a more reasonable estimation by converting from segment sizes to # of docs which makes us less prone to merges etc. Could we sample the avg doc size on disk from existing docs and make the decision ultimately based on docs?

### PR DESCRIPTION
**Pull request by dnhatn**
This is to add a `max_size` condition to the index rollover API. We use
a `totalSizeInBytes` from `DocsStats` to evaluate this condition.

Closes #27004

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/action/admin/indices/rollover/Condition.java [hunk 1]
```diff
@@ -49,6 +54,12 @@ protected Condition(String name) {
 
     public abstract Result evaluate(Stats stats);
 
+    /**
+     * Checks if this condition is available in a specific version.
+     * This makes sure BWC when introducing a new condition which is not recognized by older versions.
+     */
+    abstract boolean includedInVersion(Version version);
```
Comment (s1monw) [inline core/src/main/java/org/elasticsearch/action/admin/indices/rollover/Condition.java]: maybe just default it to `true` then you don't need to override it in the existing ones

**Review by s1monw — APPROVED** [review]
LGTM left a suggestion

File: core/src/main/java/org/elasticsearch/action/admin/indices/rollover/Condition.java (hunk 1 above)
Comment (dnhatn) [reply core/src/main/java/org/elasticsearch/action/admin/indices/rollover/Condition.java]: @s1monw, I made it `abstract` to make sure that when we add new conditions, we have to pay attention for BWC. Should I update as you suggested or keep as it is? WDYT?

Comment (dnhatn) [reply core/src/main/java/org/elasticsearch/action/admin/indices/rollover/Condition.java]: I pushed https://github.com/elastic/elasticsearch/pull/27160/commits/c33d4b35bbaa7779ddcb754a3f83b51e3d8a1f15

**Pull request merged**